### PR TITLE
sync with RomanBrodetski solidity_plonk_verifier

### PR DESCRIPTION
**Pull request by akash-chandrakar**
sync with RomanBrodetski solidity_plonk_verifier

### CODE REVIEW

File: codegen/template/Verifier.sol [hunk 1]
```diff
@@ -6,7 +6,7 @@ import "./UncheckedMath.sol";
 contract Verifier is Plonk4VerifierWithAccessToDNext {
     using UncheckedMath for uint256;
 
-    function get_verification_key() public pure returns(VerificationKey memory vk) {
+    function get_verification_key() internal pure returns(VerificationKey memory vk){
```
Comment (vladbochok) [inline codegen/template/Verifier.sol:9]: Please revert this change. It is important to have public function for getting verification keys, circuit breaker from server rely on it

File: hardhat/contracts/Verifier.sol [hunk 2]
```diff
@@ -6,7 +6,7 @@ import "./UncheckedMath.sol";
 contract Verifier is Plonk4VerifierWithAccessToDNext {
     using UncheckedMath for uint256;
 
-    function get_verification_key() public pure returns(VerificationKey memory vk) {
+    function get_verification_key() internal pure returns(VerificationKey memory vk){
```
Comment (vladbochok) [inline hardhat/contracts/Verifier.sol:9]: Same here

File: codegen/template/Verifier.sol (hunk 1 above)
Comment (akash-chandrakar) [reply codegen/template/Verifier.sol:9]: understood, just to make sure currently we are using the repo by Roman in `zksync-2-dev`  -https://github.com/matter-labs/zksync-2-dev/blob/099ce0b32011775d4639b4e52e1fb01987186a22/core/lib/types/Cargo.toml#L24
which has is at `internal` not public, so I hope it would not cause any issue.

**Comment by akash-chandrakar** [comment]
Closing this as its not needed.
